text/0000-global-allocators.md

```diff
@@ -240,9 +252,47 @@ The allocator APIs could be simplified to a more "traditional"
 malloc/calloc/free API at the cost of an efficiency loss when using allocators
 with more powerful APIs.
 
+The global allocator could be an instance of the `Allocator` trait. Since that
+trait's methods take `&mut self`, things are a bit complicated however. The
+allocator would most likely need to be a `const` type implementing `Allocator`
+since it wouldn't be sound to interact with a static. This may cause confusion
```

Comment:
I may be missing something here, but the issue is that you may not obtain a mutable reference to a static, but every method on `Allocator` takes `&mut self`:

```rust
struct MyAllocator;

impl MyAllocator {
    fn alloc(&mut self) { }
}

static ALLOCATOR: MyAllocator = MyAllocator;

fn main() {
    ALLOCATOR.alloc();
}
```

```
error: cannot borrow immutable static item as mutable
  --> <anon>:10:5
   |
10 |     ALLOCATOR.alloc();
   |     ^^^^^^^^^

error: aborting due to previous error
```

@mark-i-m There is no constructor.